### PR TITLE
Revise the documentation

### LINKED ISSUE #216: Cancellations cannot settle the computation asynchronously
As I understand it if one wants to reflect cancellation in the outcome of a future task the cancel function returned from the computation may call `resolve` or `reject` on behalf of the not-cancelled computation. However, if this is done asynchronously it has no effect on the outcome.
Quick demo:
```javascript
var futureTask = Fluture.Future(function(fail, succeed){
	const timer = setTimeout(succeed, 10000, "world");
	return function(){ 
		setTimeout(fail, 1000, "aborted!");
		clearTimeout(timer);
	};
});
var cancel = futureTask.fork(console.warn.bind(console), console.log.bind(console));
cancel(); // nothing gets logged, the forked computation never settles
```
I find this behaviour with [fun-task](https://github.com/rpominov/fun-task) as well. I figure in *fun-task* it is some kind of preventing memory leaks if a task never settles after cancellation but may hang on to the `run` i.e. `fork`-handlers. IMO that clean-up is better hooked up with `resolve` and `reject`.

Anyway, that limitation prevents building *Future* or *fun-task* wrappers for some apis, e.g. [cancelling `fetch`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController):
```javascript
var fetchJsonComputation = function(fail, succeed){
    const controller = new AbortController(); // currently just Edge and Firefox
    fetch('https://blackhole.webpagetest.org', {signal: controller.signal})
    .then(r => r.json())
    .then(succeed, fail);
    return controller.abort.bind(controller);
};
var fetchBlackholeTask = Fluture.Future(fetchJsonComputation);
// this will kick-off the ajax request
var cancel = fetchBlackholeTask.fork(console.warn.bind(console), console.log.bind(console));
// this cancels the ajax request just fine but the fork never settles with the expected DOMExecption
cancel();
```
Demo that with promises cancellation is propagated to their reject-path:
```javascript
var ctrl = new AbortController();
fetch('https://blackhole.webpagetest.org', {signal: ctrl.signal})
.then(r => r.json())
.then(console.log.bind(console), console.warn.bind(console, "Cancelled!"));
ctrl.abort(); // ~> Cancelled! DOMException
```

How do I settle a computation asynchronously in the cancel function?

### LINKED ISSUE #263: StackOverflow tag
Hello all, 

Following a previous discussion, I have added a tag for this project, called `fluture-js`. 

You may find it here:

https://stackoverflow.com/questions/tagged/fluture-js

I have also taken the liberty to add some tag information ( still under peer review ):

> What is Fluture-js ?
> From the official website:
> 
> Fluture offers a control structure similar to Promises, Tasks, Deferreds, and what-have-you. Let's call them Futures.
> 
> Much like Promises, Futures represent the value arising from the success or failure of an asynchronous operation (I/O). Though unlike Promises, Futures are lazy and adhere to the monadic interface.
> 
> Where can I get it?
> You may find more information about the project in it's official git repository:
> 
> https://github.com/fluture-js/Fluture

Please feel free to give me some feedback, suggestions and/or questions you may have.

If you are a StackOverflow fan, you may now watch the tag to be notified of any discussions that include it.

Feel free to let me know what you think, and share around ( on gitter and other channels ). 

-------

PS, if you have any corrections, now would be a good time.

### PR DESCRIPTION
**Pull request by Avaq**
* Addresses some of the minor issues people have had with the documentation
* Add a description of cancellation and creates a conceptual separation between "unsubscribing" and "cancellation". Closes #216
* Some rewording of old function descriptions.
* Broken links, indentation, and other miscellaneous fixes.
* Update the badges. Closes #263

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-[![Chat](https://badges.gitter.im/fluture-js/Fluture.svg)](https://gitter.im/fluture-js/Fluture)
-[![NPM Version](https://badge.fury.io/js/fluture.svg)](https://www.npmjs.com/package/fluture)
-[![Dependencies](https://david-dm.org/fluture-js/Fluture.svg)](https://david-dm.org/fluture-js/Fluture)
-[![Build Status](https://travis-ci.org/fluture-js/Fluture.svg?branch=master)](https://travis-ci.org/fluture-js/Fluture)
-[![Code Coverage](https://codecov.io/gh/fluture-js/Fluture/branch/master/graph/badge.svg)](https://codecov.io/gh/fluture-js/Fluture)
+[![Build Status](https://img.shields.io/travis/fluture-js/Fluture/master.svg)](https://travis-ci.org/fluture-js/Fluture)
+[![Code Coverage](https://img.shields.io/codecov/c/github/fluture-js/Fluture/master.svg)](https://codecov.io/gh/fluture-js/Fluture/branch/master)
+[![NPM Downloads](https://img.shields.io/npm/dm/fluture.svg)](https://www.npmjs.com/package/fluture)
+[![Gitter Chat](https://img.shields.io/gitter/room/fluture-js/Fluture.svg)](https://gitter.im/fluture-js/Fluture)
+[![StackExchange](https://img.shields.io/stackexchange/stackoverflow/t/fluture.svg)](https://stackoverflow.com/questions/tagged/fluture)
```
Comment (davidchambers) [inline README.md]: This image is not loading for me.

File: README.md [hunk 2]
```diff
@@ -97,15 +96,17 @@ This script will add `Fluture` to the global scope.
...
 ## Butterfly
 
 The name "Fluture" is a conjunction of "FL" (the acronym to [Fantasy Land][FL])
 and "future". Fluture means butterfly in Romanian: A creature you might expect
 to see in Fantasy Land.
 
-Thanks to [Erik Fuente][8] for the logo, and [WEAREREASONABLEPEOPLE][9] for
-sponsoring the project.
+Credits go to Erik Fuente for styling the logo, and [WEAREREASONABLEPEOPLE][9]
+for sponsoring the project.
```
Comment (davidchambers) [inline README.md]: Credit is a mass noun in this context, so I suggest dropping the “s”.

File: README.md [hunk 3]
```diff
@@ -199,35 +204,36 @@ sponsoring the project.
 
 ### Type signatures
 
-[Hindley-Milner][Guide:HM] type signatures are used to document functions. You
-might encounter some additional syntax that we use to describe JavaScript
-specific stuff, like [methods](#squiggly-arrows) or functions that take
+The various function signatures are provided in a small language referred to as
+Hindley-Milner notation. You can learn about it in the context of JavaScript
+[here][Guide:HM]. On top of the basic Hindley-Milner notation, we use a few
```
Comment (davidchambers) [inline README.md]: Link text should be descriptive. I suggest rewording this so the link text is “Hindley–Milner notation” or similar.

File: README.md [hunk 4]
```diff
@@ -774,11 +854,15 @@ Future.prototype.chain :: Future e a ~> (a -> Future e b) ->        Future e b
...
-new Future will not be created until the other one is forked. The function is
-only ever applied to the resolution value; it's ignored when the Future was
-rejected. To learn more about the exact behaviour of `chain`, check out its [spec][FL:chain].
+Sequence a new Future using the resolution value from another. Similarly to
+[`map`](#map), `chain` expects a function to transform the resolution value of
+a Future. But instead of returning the new *value*, chain expects a Future to
+be returned.
+
+The transformation is only applied to the resolution branch: If the Future is
+rejected, the transformation is ignored.
```
Comment (davidchambers) [inline README.md]: I suggest replacing “If” with “if” after the colon.

File: README.md (hunk 1 above)
Comment (Avaq) [reply README.md]: Yeah, it appears that the stackoverflow api might be rate liited. The image only loads very rarely for me, and on https://shields.io/ the badge shows as "vendor unesponsive":

latest-screenshot

I have seen it working a few times though, by zealously refreshing the page. It might not be worth including in the docs if it rarely loads, though. However, when the image doesn't load, the `alt` is rendered ("StackExchange"). This text still has some value, and links to the appropriate section on StackExchange.

Comment (davidchambers) [reply README.md]: I don't think we should include an unreliable badge. We could provide a text link instead.

> However, when the image doesn't load, the `alt` is rendered ("StackExchange").

I don't see the alt text in Safari on macOS.

Comment (Avaq) [reply README.md]: Safari is the new Internet Explorer.

Anyway, what do you think of rendering something like this instead?

Community resources | Badges
--------------------|------
[Build Status][]    | ![Build Status Badge][]
[Code Coverage][]   | ![Code Coverage Badge][]
[Dependencies][]    | ![Greenkeeper Badge][] ![David DM Badge][]
[NPM Package][]     | ![NPM Version Badge][] ![NPM Downloads Badge][]
[Gitter Chat][]     | ![Gitter Chat Badge][]
[StackOverflow][]   | ![StackOverflow Questions Badge][]

[Build Status]: https://travis-ci.org/fluture-js/Fluture
[Code Coverage]: https://codecov.io/gh/fluture-js/Fluture/branch/master
[Dependencies]: https://david-dm.org/fluture-js/Fluture
[NPM Package]: https://www.npmjs.com/package/fluture
[Gitter Chat]: https://gitter.im/fluture-js/Fluture
[StackOverflow]: https://stackoverflow.com/questions/tagged/fluture
[Build Status Badge]: https://img.shields.io/travis/fluture-js/Fluture/master.svg
[Code Coverage Badge]: https://img.shields.io/codecov/c/github/fluture-js/Fluture/master.svg
[Greenkeeper Badge]: https://badges.greenkeeper.io/fluture-js/fluture-retry.svg
[David DM Badge]: https://img.shields.io/david/fluture-js/Fluture.svg
[NPM Version Badge]: https://img.shields.io/npm/v/fluture.svg
[NPM Downloads Badge]: https://img.shields.io/npm/dm/fluture.svg
[Gitter Chat Badge]: https://img.shields.io/gitter/room/fluture-js/Fluture.svg
[StackOverflow Questions Badge]: https://img.shields.io/stackexchange/stackoverflow/t/fluture.svg

File: README.md (hunk 3 above)
Comment (Avaq) [reply README.md]: :zap:

File: README.md (hunk 4 above)
Comment (Avaq) [reply README.md]: :zap:

File: README.md (hunk 1 above)
Comment (Avaq) [reply README.md]: Everything just loaded, so I took a quick screenshot before it was gone! 'Gave me Pokemon Snap flashbacks! That's what it looks like:

latest-screenshot

Comment (davidchambers) [reply README.md]: The table looks fine, but I don't think it belongs at the beginning of the document. I lean towards sticking with a single row of badges.

Comment (Avaq) [reply README.md]: Some of these badges are not informative, and are supposed to be used as links. That's where this approach collapses. I'm not sure how to proceed.

Comment (Avaq) [reply README.md]: > I lean towards sticking with a single row of badges.

Agreed. Alright. I will just pretend that all browsers show the alt text, and treat the badges like menu items. Who uses Safari anyway? ;)
